tests/test_e2e/test_cloud_spec.py

```diff
@@ -0,0 +1,58 @@
+import ops
+import pytest
+
+import scenario
+
+
+class MyCharm(ops.CharmBase):
+    def __init__(self, framework: ops.Framework):
+        super().__init__(framework)
+        for evt in self.on.events().values():
+            self.framework.observe(evt, self._on_event)
+
+    def _on_event(self, event):
+        pass
+
+
+def test_get_cloud_spec():
+    scenario_cloud_spec = scenario.CloudSpec(
+        type="lxd",
+        name="localhost",
+        endpoint="https://127.0.0.1:8443",
+        credential=scenario.CloudCredential(
+            auth_type="clientcertificate",
+            attributes={
+                "client-cert": "foo",
+                "client-key": "bar",
+                "server-cert": "baz",
+            },
+        ),
+    )
+    expected_cloud_spec = ops.CloudSpec(
+        type="lxd",
+        name="localhost",
+        endpoint="https://127.0.0.1:8443",
+        credential=ops.CloudCredential(
+            auth_type="clientcertificate",
```

Comment:
side-note: should 'clientcertificate' be the default value of `auth_type`? or did you make it up?

Reply:
This is only a test, I made it up. There is no default type I guess, it depends on the cloud?